### ISSUE TITLE
Cloak should disengage when running out of fuel

### ISSUE THREAD

**Issue by UnciaPrima** [endless-sky/endless-sky#3398]
Well, the title says everything.

Right now, when you run out of fuel while cloaked, you become visible again (and that's the intended behavior), but fuel does continue to drain. I think this could be improved by auto-disengaging the cloaking device if you run out of fuel.

Thanks for reading
UP

**Comment by Bladewood** [comment]
Partially done by https://github.com/endless-sky/endless-sky/commit/d754ba6e2f7ccd105ab564c04445e69fb32fafc4, though fully canceling the cloaking action isn't implemented, as far as I know.

**Comment by tehhowch** [comment]
What's the issue here? Having the cloak command issued only drains the associated fuel and energy when the ship is able to actually activate it's cloaking tech - not when it is finally fully cloaked. As a player, part of your job as (a fleet) captain is to manage advanced ship functions such as deploying fighters or attempting to cloak. 

Disengaging the player's cloak command is not quite as simple as it sounds, because it is a fleet-wide command. Are your Starlings really supposed to drop their cloak just because your flagship ran out of fuel? What if the flagship doesn't have the ability to cloak? Should the flagship's cloak be turned off when your escort runs out of fuel for its cloak?

**Comment by UnciaPrima** [comment]
> What's the issue here? Having the cloak command issued only drains the associated fuel and energy if the ship is able to actually activate it's cloaking tech - not just when it is finally fully cloaked. As a player, part of your job as (a fleet) captain is to manage advanced ship functions such as deploying fighters or attempting to cloak.

I must have badly explained the issue. I'm not talking about the "transition" time between *not cloaked* and *fully cloaked* and have no problem with the current behavior.

Instead I am talking about what happens when you do the following:
- Have one ramscoop equipped
- Spend most of your fuel by jumping or else
- Engage your cloak until it consumes all the remaining fuel

Then, when there is no more fuel, the cloaking device *seemingly* disengages (as you become visible again) but your fuel does not go up. You have to manually deactivate your cloak in order to begin collecting fuel from your ramscoop again. This makes no sense, as your cloaking device should not consume any energy if it is not doing anything. This is the issue.

**Comment by KiLEdEnNis** [comment]
that's probably because your ship keeps trying to recloak and disengages because it runs out immediately after

**Comment by tehhowch** [comment]
>your ship keeps trying to recloak and disengages because it runs out immediately after

Precisely.

If you return to the system center, where you may have sufficient fuel recovery to fully power your cloaking tech's fuel consumption, you will sustain the activation of your cloaking technology for long enough that you can fully cloak (or at minimum, longer, because more fuel was collected while your cloak was in a recovery/disrupted state).

**Comment by UnciaPrima** [comment]
Yes that is certainly what's happening. I'm definitely not saying it's a bug. What I'm saying is that it might not be the best way to treat that specific situation.
In 99.9% of the cases, you're better off with a fully disengaged cloak, waiting for your fuel to go up a little, that with a cloak soaking up all your fuel while achieving absolutely nothing.

**Comment by tehhowch** [comment]
The specific situation is a judgement call between resources on your flagship and resources on your escorts. The game shouldn't make every decision for the player. It can certainly be altered to make such a decision, but as I mention above, the only guarantee that the correct choice is made is that the player makes the choice (because there is no "correct" choice).

**Comment by UnciaPrima** [comment]
> The specific situation is a judgement call between resources on your flagship and resources on your escorts. The game shouldn't make every decision for the player. It can certainly be altered to make such a decision, but as I mention above, the only guarantee that the correct choice is made is that the player makes the choice (because there is no "correct" choice).

Well, if, as you say, "Disengaging the player's cloak command is not quite as simple as it sounds, because it is a fleet-wide command", then maybe the real issue here is that the game should indeed allow each ship to disengage separately from the other ones in your fleet, in order to optimize energy.

**Comment by Tadrix** [comment]
How about having ability to command only certain ships to cloak, like it does with moving? Else, if they decloak on their own, there would not be a way to recloak without decloaking whole fleet first.

**Comment by KiLEdEnNis** [comment]
selecting a group of ships and having cloak/deploy fighters only apply to them could work

**Comment by UnciaPrima** [comment]
What about simply using the key for your flagship, and shift-key for you fleet (or the selected ships of course)? It would fit the general keyboard behavior and would allow for auto-disengaging your own cloak when out of fuel.
What do you think?
